### ISSUE TITLE
Potentially misleading note in documentation for PackedSequence

### ISSUE THREAD

**Issue by Dris101** [pytorch/pytorch#59629]
## Issue

The notes for the `PackedSequence` class 
[https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.PackedSequence.html#torch.nn.utils.rnn.PackedSequence](https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.PackedSequence.html#torch.nn.utils.rnn.PackedSequence) state

> Instances of this class should never be created manually. They are meant to be instantiated by functions like pack_padded_sequence().

It seems however that there are instances where creating them manually is (I think) unavoidable. For example I have a use case in RL where I am passing variable length sequences of images through a conv net and then passing the output to an LSTM layer. The easiest thing to do here is to pack the padded sequences of input images with `pack_padded_sequence` and then pass the `.data` field of the resulting `PackedSequence` through the conv net before reconstituting a `PackedSequence` with the output of the conv net and passing that as an argument to the recurrent layer. Indeed this is a technique recommended in other issues and the forums. For example in [https://github.com/pytorch/pytorch/issues/25310]() @andreaskoepf elegantly suggests creating a function such as
```
def squash_packed(x, fn=torch.tanh):
    return torch.nn.utils.rnn.PackedSequence(fn(x.data), x.batch_sizes, 
                 x.sorted_indices, x.unsorted_indices)
```
to map the first `PackedSequence` to the second by applying a given function. The critical thing is that the `.batch_sizes, .sorted_indices` and `.unsorted_indices` fields from the original `PackedSequence` output by `pack_padded_sequence` must be preserved and used when manually creating the new one. Given the instruction not to create them manually in the note I struggled for a while to work out how to do this.

## Suggested documentation revision

Remove the stipulation that `PackedSequences` "should never be created manually" and instead include something like the above with caveats on how to do it safely.

## Possible enhancement suggestion

Add a mapping function such as the above to the `PackedSequence` class to abstract the details away from the user in case there are any other potential pitfalls to be avoided.

**Comment by Dris101** [comment]
Agreed. Seems `NestedTensors` are making progress now, though I expect `PaddedSequence` will be with us for a while yet! I think just softening the docs language in the note as you suggest makes sense.

**Comment by uditrana** [comment]
This might not be the right place, but I have a similar use case as OP (variable sequence length of images that go through conv layers before being fed into a LSTM). 

I tried various "hacks" by using the packed_sequence.data as inputs to the conv layers but that causes mixed type problems. Is the current canonical approach to handling this using the squash_packed above? How do we "trick" the conv layers to only operate on the unpadded data, but pass a correct packed_sequence to the RNN components?
